table/block_based/block_based_table_factory.cc

```diff
@@ -576,7 +576,8 @@ Status BlockBasedTableFactory::NewTableReader(
       table_reader_options.block_cache_tracer,
       table_reader_options.max_file_size_for_l0_meta_pin,
       table_reader_options.cur_db_session_id, table_reader_options.cur_file_num,
-      table_reader_options.unique_id);
+      table_reader_options.unique_id,
+      table_reader_options.user_defined_timestamps_persisted);
```

Comment:
That's a good question. I'm working on adding a per-SST file marker in the manifest for the user key format contained in each file and the `smallest` and `largest` in file metadata. Similar to the timestamp size marker in WAL logs.  I'm still sorting out how to make it compatible with users who are already using user-defined timestamps.

Reply:
Would an optional "timestamps NOT persisted" field in `AddFile` (`kNewFile4` record) work? I.e. the default would be "timestamps persisted".

Reply:
Thanks for the idea, this optional flag is indeed the best approach. One scenario I'm thinking about is when switching user comparator is supported and an existing column family that was not using UDT before start to use UDT and just retain timestamps in memtables. For this column family, it has SST files generated prior to this switch, that have this optional flag interpreted as `True`, and it will have SST files generated after this switch with this flag explicitly marked as `False`. In both cases, these SST files do not contain user-defined timestamps, and padding is needed before the running comparator can handle these user keys. 

I think maybe we can enforce some naming rules for a user comparator with non-zero timestamp size, like it needs to include the keyword `ts` in its name, etc. And use this optional flag together with the existing comparator name entry in the Manifest to handle cases like this. This assumes for users that are already using this experimental feature, they are only using the convenient ts-aware comparators provided by RocksDB, not implementing their customized comparator. What do you think?

Reply:
The great thing about the manifest is that we rewrite it all the time. So I think when we detect the comparator switch, we can just force-create a new manifest file and mark any old files as "timestamps not persisted" there and then (as if they had been written using UDTS-in-memtables-only).

Reply:
ah, I see, I didn't know that. It makes a lot of sense that Manifest file can be rewritten since it's manifest. Thanks a lot for the idea! I will look into it more. This is a very simple and elegant way to handle the case.